### PR TITLE
Added cache_dir and pass_through_pattern

### PR DESCRIPTION
**Pull request by elfranne**
Set the location of the cache directory where all the deb will be stored.
Set the pass_through_pattern to allow SSL repo to bypass the cache.

### CODE REVIEW

**Comment by elfranne** [comment]
There are a few good examples for the pass_through_patterns [here](https://blog.packagecloud.io/eng/2015/05/05/using-apt-cacher-ng-with-ssl-tls/)

The file zzz_override.conf is due to a 'new feature' see bug report [here](https://bugs.launchpad.net/ubuntu/+source/apt-cacher-ng/+bug/1279021)

File: manifests/config.pp [hunk 1]
```diff
@@ -21,4 +23,16 @@
         require => Class['aptcacherng::install'],
         notify  => Class['aptcacherng::service'],
     }
+
+
+    file { 'zzz_override.conf':
+        ensure  => present,
+        name    => $::aptcacherng::params::config_name,
```
Comment (mattock) [inline manifests/config.pp]: This name is a duplicate, see above.

File: manifests/config.pp [hunk 2]
```diff
@@ -6,7 +6,9 @@
 class aptcacherng::config
 (
     $listen_addresses,
-    $port
+    $port,
+    $cache_dir,
```
Comment (mattock) [inline manifests/config.pp]: Having data types for $cache_dir and $pass_through_pattern would be nice.

File: templates/acng.conf.erb [hunk 3]
```diff
@@ -293,6 +293,7 @@ ExTreshold: 4
 # Examples: 
 # PassThroughPattern: private-ppa\.launchpad\.net:443$ 
 # PassThroughPattern: .* # this would allow CONNECT to everything
+PassThroughPattern: <%= @pass_through_pattern %>
```
Comment (mattock) [inline templates/acng.conf.erb:296]: Is ```PassThroughPattern: .*``` the implicit default?

**Comment by elfranne** [comment]
@mattock ?

**Comment by mattock** [comment]
@elfranne I made some inline comments. I believe there is resource title clash that needs to be fixed. Can you verify that?

**Comment by elfranne** [comment]
@mattock Seems your comments got lost or something (can't see any comments) ...

**Comment by elfranne** [comment]
@mattock  ?

**Comment by mattock** [comment]
@elfranne sorry about the delay. For some reason people don't seem to see my inline comments... I need to debug that.

So three things:

* The value of "name" parameter in zzz_override.conf is a duplicate ($::aptcacherng::params::config_name)
* I also asked if "PassThroughPattern: .*" is the implicit default?
* If you can/want, please add data types to the parameters. I won't mind if you add it even in places that your code does not touch :smile:.

**Comment by elfranne** [comment]
PassThroughPattern: It's the value that will allow you to proxy https repos but without caching it (because it's encrypted). There is no default value, but have a look on the relevant part of the [docs](https://www.unix-ag.uni-kl.de/~bloch/acng/html/howtos.html#ssluse) 

Also not sure if listen_addresses has the best format as it is complicated to validate the input. How about an array with different types ? localhost + ip + fqdn ?

**Comment by elfranne** [comment]
https://github.com/simp/puppetlabs-stdlib/#stdlibhost
Something like : 
Array[Stdlib::Host] listen_addresses = ['localhost']

**Comment by mattock** [comment]
Looks pretty good and good call on StdLib data types. Can you add stdlib to metadata,json? Besides that I think this is ready to be merged.

**Review by mattock — APPROVED** [review]

**Comment by elfranne** [comment]
There's no Changelog ?
Also I haven't bumped any version, I ll let you do that when you do the release.

**Comment by mattock** [comment]
For Puppet Forge having a Changelog would probably make sense. But there isn't one yet.

**Pull request merged**